File: ems-back/front/ims-frontend/src/service/EmployeeService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/employee';

export const listEmployees = () => axios.get(API_URL);

export const createEmployee = (employee) => axios.post(API_URL, employee);

export const getEmployee = (employeeId) => axios.get(API_URL + '/' + employeeId);

export const updateEmployee = (employeeId, employee) => axios.put(API_URL + '/' + employeeId, employee);

export const deleteEmployee = (employeeId) => axios.delete(API_URL + '/' + employeeId);

File: ems-back/src/main/java/com/fs/emsback/service/EmployeeService.java
package com.fs.emsback.service;

import com.fs.emsback.entity.dto.EmployeeDTO;

import java.util.List;

public interface EmployeeService {
    EmployeeDTO createEmployee(EmployeeDTO employeeDTO);

    EmployeeDTO getEmployee(Long EmployeeId);

    List<EmployeeDTO> getAllEmployees();

    EmployeeDTO updateEmployee(Long employeeId, EmployeeDTO upEmployee);

    void deleteEmployee(Long employeeId);
}
